lib/cdo/delete_accounts_helper.rb

```diff
@@ -98,6 +98,17 @@ def clean_and_destroy_pd_content(user_id)
       application.notes = nil
       application.save! validate: false
     end
+
+    # Two different paths to anonymizing attendance records
+    Pd::Attendance.with_deleted.where(teacher_id: user_id).each do |attendance|
+      attendance.destroy!
+      attendance.update!(teacher_id: nil)
+    end
+    Pd::Attendance.with_deleted.where(marked_by_user_id: user_id).each do |attendance|
+      attendance.destroy!
+      attendance.update!(marked_by_user_id: nil)
+    end
```

Comment:
Are we sure we want to destroy the attendance in the marked_by case? That field tracks who marked the teacher in attendance. In the typical case it will be the teacher, but it could also be overridden by the facilitator or a Code.org workshop admin. If that admin or facilitator (no the attending teacher) deletes their account I think we'd want to clear the `marked_by` field so we lose track that that particular user marked the teacher in attendance, but the teacher's attendance should still be counted... at least unless they delete their account.

Reply:
That's a good question.  I suppose we don't actually need to soft-delete the attendance in this case because the record is still valid without a `marked_by_user` on file.  I'll change the behavior.